libmcount/plthook.c

```diff
@@ -242,8 +243,10 @@ static int find_got(struct uftrace_elf_data *elf,
 			pd->module_id = (unsigned long)pd;
 		}
 
-		pr_dbg2("found GOT at %p (PLT resolver: %#lx)\n",
-			pd->pltgot_ptr, plthook_resolver_addr);
+		pr_dbg2("found GOT at %p (file_addr: %#lx, PLT resolver: %#lx)\n",
+			pd->pltgot_ptr,
+			(unsigned long)pd->pltgot_ptr - pd->base_addr,
+			plthook_resolver_addr);
```

Comment:
Now I think that this is somewhat duplicate.  How about just display the resolver address only?

Reply:
I didn't clearly understand what `plthook_resolver_addr` mean. As far as I know so far, it's a function that resolves the library address and it's in `ld-linux.*.so` library. But didn't clearly examine its behavior.

What I understand from `pd->pltgot_ptr` is that it's the address of _GLOBAL_OFFSET_TABLE_ actually loaded on memory. But it was difficult to compare the address to the output of `objdump -D` because of the offset, so added the the second info for easier comparison. It doesn't seem to be duplicated for me.

Reply:
I mean that GOT address was printed from above (another `pr_dbg2`).

Reply:
It seems that the first `pr_dbg2` contains too much info.
How about this compared to the previous change?
```diff
diff --git a/libmcount/plthook.c b/libmcount/plthook.c
index dc632315..1cc1c076 100644
--- a/libmcount/plthook.c
+++ b/libmcount/plthook.c
@@ -220,9 +220,10 @@ static int find_got(struct uftrace_elf_data *elf,
        pd->base_addr  = offset;
        pd->plt_addr   = plt_addr;

-       pr_dbg2("\"%s\" is loaded at %#lx (id: %#lx, PLTGOT: %p)\n",
-               basename(pd->mod_name), pd->base_addr, pd->module_id,
-               pd->pltgot_ptr);
+       pr_dbg2("\"%s\" is loaded at %#lx (id: %#lx)\n",
+               basename(pd->mod_name), pd->base_addr, pd->module_id);
+       pr_dbg2("GOT is located at %p (file_addr: %#lx)\n", pd->pltgot_ptr,
+               (unsigned long)pd->pltgot_ptr - pd->base_addr);

        memset(&pd->dsymtab, 0, sizeof(pd->dsymtab));
        load_elf_dynsymtab(&pd->dsymtab, elf, pd->base_addr, SYMTAB_FL_DEMANGLE);
@@ -243,9 +244,7 @@ static int find_got(struct uftrace_elf_data *elf,
                        pd->module_id = (unsigned long)pd;
                }

-               pr_dbg2("found GOT at %p (file_addr: %#lx, PLT resolver: %#lx)\n",
-                       pd->pltgot_ptr,
-                       (unsigned long)pd->pltgot_ptr - pd->base_addr,
+               pr_dbg2("found GOT.  address of PLT resolver: %#lx\n",
                        plthook_resolver_addr);

                restore_plt_functions(pd);
```

Reply:
The module_id is related to GOT so I think it's better moving it to the "found GOT" line.

Reply:
How about this then?
```c
        if (plt_found) {
                if (plthook_resolver_addr == 0)
                        plthook_resolver_addr = pd->pltgot_ptr[2];

                pr_dbg2("module id is %#lx\n", pd->module_id);
                if (pd->module_id == 0) {
                        pr_dbg2("update module id to %p\n", pd);
                        overwrite_pltgot(pd, 1, pd);
                        pd->module_id = (unsigned long)pd;
                }

                pr_dbg2("found GOT at %p (file_vaddr: %#lx)\n", pd->pltgot_ptr,
                        (unsigned long)pd->pltgot_ptr - pd->base_addr);
                pr_dbg2("PLT resolver address: %#lx\n", plthook_resolver_addr);

                restore_plt_functions(pd);
        }
```

Reply:
If you don't like this, then it'd be helpful if you can modify it as you like. Thanks.

Reply:
What I have in mind is:

```c

pr_dbg2("\"%s\" is loaded at %#lx\n");

if (plt_found) {
    pr_dbg2("found GOT at %p (file_vaddr: %#lx)\n");
    pr_dbg2("module id = %#lx, PLT resolver = %#lx\n");
    ...

Reply:
Thanks. I just updated it.